src/contexts/ConfigContext.tsx

```diff
@@ -0,0 +1,39 @@
+import { ReactNode, createContext, useContext, useState, FC } from 'react';
+import defaultConfig from '@/config';
+const ConfigContext = createContext({});
+
+interface ConfigProvider {
+  config?: Config,
+  setConfig?: () => void,
+  openedPanels?: Panel[],
+  setOpenedPanels?: (panels: Panel[]) => void
+}
+
+interface ConfigProviderProps {
+  children?: ReactNode
+}
+
+const ConfigProvider: FC<ConfigProviderProps> = ({ children }) => {
+  const [config, setConfig]  = useState <Config>(defaultConfig);
+  const [openedPanels, setOpenedPanels] = useState(defaultConfig.panels);
+
+  function updateConfig(newConfig: Config) {
+    setConfig(newConfig);
+  }
+
+  return (
+    <ConfigContext.Provider
+      value={{ config, updateConfig, openedPanels, setOpenedPanels }}
+    >
+      {children}
+    </ConfigContext.Provider>
+  );
+};
+
+function useConfig(): ConfigProvider {
+  return useContext(ConfigContext)
+}
+
+
+
+export { ConfigProvider, useConfig };
```

Comment:
The export gives me a fast fresh ESLint error. Since the official React [example](https://codesandbox.io/p/sandbox/react-dev-59fz7l?file=%2Fsrc%2FTasksContext.js&utm_medium=sandpack) shows to use it in one file I suggest to add supress the `react-refresh/only-export-components` ESlint rule.

Reply:
In my code eslint does not raise any error.. Is it still the case for you ?